src/CarbonAware.CLI/test/unitTests/Commands/EmissionsCommandTests.cs

```diff
@@ -0,0 +1,167 @@
+            "{" + $"{expectedLocationJSON},{expectedTimeJSON},{expectedDurationJSON},{expectedRatingJSON}" + "}")!;
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .ReturnsAsync(new List<EmissionsData>() { emissionsData });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
+
+        // Assert
+        StringAssert.Contains(expectedLocationJSON, _console.Out.ToString());
+        StringAssert.Contains(expectedTimeJSON, _console.Out.ToString());
+        StringAssert.Contains(expectedDurationJSON, _console.Out.ToString());
+        StringAssert.Contains(expectedRatingJSON, _console.Out.ToString());
+        _mockCarbonAwareAggregator.Verify(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()), Times.Once);
+    }
+
+    [Test]
+    public async Task Run_CallsAggregatorWithLocationOptions()
+    {
+        // Arrange
+        var emissionsCommand = new EmissionsCommand();
+        var longAliasLocation = "eastus";
+        var shortAliasLocation = "westus";
+        var invocationContext = SetupInvocationContext(emissionsCommand, $"emissions --location {longAliasLocation} -l {shortAliasLocation}");
+        var expectedLocations = new List<string>() { longAliasLocation, shortAliasLocation };
+        IEnumerable<string> actualLocations = Array.Empty<string>();
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .Callback((CarbonAwareParameters _parameters) => {
+                actualLocations = _parameters.MultipleLocations.Select(l => l.DisplayName);
+             });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
+
+        // Assert
+        _mockCarbonAwareAggregator.Verify(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()), Times.Once);
+        CollectionAssert.AreEquivalent(expectedLocations, actualLocations);
+    }
+
+    [TestCase("--start-time", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run StartTimeOption: long alias")]
+    [TestCase("-s", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run StartTimeOption: short alias")]
+    public async Task Run_CallsAggregatorWithStartTimeOptions(string alias, string optionValue)
+    {
+        // Arrange
+        var emissionsCommand = new EmissionsCommand();
+        var invocationContext = SetupInvocationContext(emissionsCommand, $"emissions {alias} {optionValue}");
+        var expectedStartTime = DateTimeOffset.Parse(optionValue);
+        DateTimeOffset actualStartTime = default;
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .Callback((CarbonAwareParameters _parameters) => {
+                actualStartTime = _parameters.Start;
+            });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
```

Comment:
Shouldn't the command fail if there is no location parameter specified?

Reply:
In a real call, yes, the command is parsed and validated, and only if valid is the `Run` handler invoked.  In our tests, since we call `Run` directly we can assert individual options without bothering with setting up "required" options not under test.